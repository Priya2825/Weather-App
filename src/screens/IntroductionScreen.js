import React from "react";
import {
  View,
  Text,
  StyleSheet,
  Image,
  TouchableOpacity,
  Dimensions,
} from "react-native";
import { LinearGradient } from "expo-linear-gradient";
import icon from "../../images/introImage.png";
import IntroductionCarousel from "../../components/IntroductionCarousel";
import "react-native-screens";
import "react-native-safe-area-context";
import MaterialCommunityIcons from "react-native-vector-icons/MaterialCommunityIcons";

const screenWidth = Dimensions.get("screen").width;
const screenHeight = Dimensions.get("screen").height;

function IntroductionScreen({ onFinish }) {
  return (
    <View style={styles.container}>
      <LinearGradient
        testID="background-gradient"
        colors={["#8EC5FC", "#E0C3FC"]}
        style={styles.background}
      >
        <View style={styles.imageContainer}>
          <Image testID="intro-image" source={icon} style={styles.tinyLogo} />
        </View>
        <View style={styles.carouselContainer} testID="introduction-carousel">
          <IntroductionCarousel />
        </View>
        <TouchableOpacity
          testID="continue-button"
          onPress={onFinish}
          style={styles.buttonContainer}
        >
          <View style={styles.circleButton}>
            <MaterialCommunityIcons
              name="arrow-right"
              color={"#fff"}
              size={40}
            />
          </View>
        </TouchableOpacity>
      </LinearGradient>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: "#FFF", // Background color in case the gradient fails to load
  },
  background: {
    position: "absolute",
    left: 0,
    right: 0,
    top: 0,
    height: screenHeight,
    justifyContent: "space-between",
    alignItems: "center",
  },
  tinyLogo: {
    height: 350,
    width: 350,
    objectFit: "contain",
  },
  carouselContainer: {
    height: screenHeight / 2.5,
    width: screenWidth,
  },
  imageContainer: {
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    height: screenHeight - screenHeight / 2.5,
  },
  buttonContainer: {
    position: "absolute",
    bottom: 50,
    alignItems: "center",
    justifyContent: "center",
  },
  circleButton: {
    width: 60,
    height: 60,
    backgroundColor: "#4CAF50",
    borderRadius: 30,
    justifyContent: "center",
    alignItems: "center",
    elevation: 5,
    shadowColor: "#000",
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.3,
    shadowRadius: 3,
  },
  arrow: {
    fontSize: 28,
    color: "#FFF",
  },
});

export default IntroductionScreen;
