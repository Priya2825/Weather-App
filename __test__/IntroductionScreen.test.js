import React from "react";
import { render, screen, fireEvent } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import IntroductionScreen from "../src/screens/IntroductionScreen";

// Snapshot Test
describe("IntroductionScreen", () => {
  it("introduction screen renders correctly", () => {
    const tree = renderer
      .create(<IntroductionScreen onFinish={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

// Rendering and Component Verification
describe("IntroductionScreen Rendering", () => {
  it("renders all essential components", () => {
    render(<IntroductionScreen onFinish={() => {}} />);

    // Verify the gradient background
    expect(screen.getByTestId("background-gradient")).toBeTruthy();

    // Verify the image
    const image = screen.getByTestId("intro-image");
    expect(image).toBeTruthy();
    expect(image.props.source).toEqual(require("../images/introImage.png"));

    // Verify the carousel
    expect(screen.getByTestId("introduction-carousel")).toBeTruthy();

    // Verify the button
    const button = screen.getByTestId("continue-button");
    expect(button).toBeTruthy();
  });
});

// Interaction Test
describe("IntroductionScreen Interactions", () => {
  it("calls onFinish when the button is pressed", () => {
    const onFinishMock = jest.fn();
    render(<IntroductionScreen onFinish={onFinishMock} />);

    const button = screen.getByTestId("continue-button");
    fireEvent.press(button);

    expect(onFinishMock).toHaveBeenCalled();
  });
});
