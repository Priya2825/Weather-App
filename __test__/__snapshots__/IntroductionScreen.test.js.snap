// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IntroductionScreen Snapshot renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    colors={
      [
        4287546876,
        4292920316,
      ]
    }
    style={
      {
        "alignItems": "center",
        "height": 1334,
        "justifyContent": "space-between",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    testID="background-gradient"
  >
    <View
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "height": 800.4,
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={1}
        style={
          {
            "height": 350,
            "objectFit": "contain",
            "width": 350,
          }
        }
        testID="intro-image"
      />
    </View>
    <View
      style={
        {
          "height": 533.6,
          "width": 750,
        }
      }
      testID="introduction-carousel"
    >
      <RNCViewPager
        initialPage={0}
        layoutDirection="ltr"
        onMoveShouldSetResponderCapture={[Function]}
        onPageScroll={[Function]}
        onPageScrollStateChanged={[Function]}
        onPageSelected={[Function]}
        style={
          {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "flex": 1,
                  "justifyContent": "center",
                  "padding": 20,
                  "paddingBottom": 60,
                },
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
          >
            <Text
              style={
                {
                  "color": "#0000FF",
                  "fontSize": 20,
                  "fontWeight": "600",
                  "letterSpacing": 1.2,
                  "lineHeight": 28,
                  "textAlign": "center",
                }
              }
            >
              Welcome to WeatherPro
            </Text>
            <Text
              style={
                {
                  "color": "#0000FF",
                  "fontSize": 16,
                  "letterSpacing": 1,
                  "lineHeight": 22,
                  "marginTop": 10,
                  "textAlign": "center",
                }
              }
            >
              Swipe to continue
            </Text>
          </View>
        </View>
        <View
          collapsable={false}
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "flex": 1,
                  "justifyContent": "center",
                  "padding": 20,
                  "paddingBottom": 60,
                },
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
          >
            <Text
              style={
                {
                  "color": "#0000FF",
                  "fontSize": 20,
                  "fontWeight": "600",
                  "letterSpacing": 1.2,
                  "lineHeight": 28,
                  "textAlign": "center",
                }
              }
            >
              Stay ahead of the weather with accurate forecasts and personalized alerts.
            </Text>
          </View>
        </View>
        <View
          collapsable={false}
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "flex": 1,
                  "justifyContent": "center",
                  "padding": 20,
                  "paddingBottom": 60,
                },
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
          >
            <Text
              style={
                {
                  "color": "#0000FF",
                  "fontSize": 20,
                  "fontWeight": "600",
                  "letterSpacing": 1.2,
                  "lineHeight": 28,
                  "textAlign": "center",
                }
              }
            >
              Weather updates for your specific location
            </Text>
          </View>
        </View>
      </RNCViewPager>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "bottom": 50,
          "justifyContent": "center",
          "opacity": 1,
          "position": "absolute",
        }
      }
      testID="continue-button"
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#4CAF50",
            "borderRadius": 30,
            "elevation": 5,
            "height": 60,
            "justifyContent": "center",
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 3,
            "width": 60,
          }
        }
      >
        <Text />
      </View>
    </View>
  </ViewManagerAdapter_ExpoLinearGradient>
</View>
`;
